keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,375 @@
+Primary Interface of the Pod.
+
+<p align="center">
+  <img src="mn-story-8.png?raw=true" alt="multi-network story 8 network interface hot-plug"/>
+</p>
+
+#### Story #9
+As an infrastructure administrator I wish to support IPv6 SLAAC in the
+Kubernetes Pods. SLAAC's nature is to dynamically assign IP addresses and change
+them over time. Those should be reflected in Pod status. This would not be
+applicable for the Primary Interface of the Pod.
+
+### Requirements
+
+Below requirements are divided into phases. Each phase will produce a separate
+KEP with detailed design for specified set of requirements. Each of this KEP
+will have to take all of the requirements, from each phase, into consideration
+when created.
+
+#### Phase I (base API and reference in Pod)
+1. This effort shall not change the behavior of Today existing clusters
+2. We need to introduce an “Object” that represents the existing 
+infrastructure’s networking
+3. “Object” is decoupled from the network definition/description created by the
+network administrator
+4. The “Object” shall not define any implementation specific parameters in that
+object
+5. “Object” shall provide option to define:
+   * IPAM mode: external/internal
+   * List of route prefixes - optional and not forced on the implementations
+6. “Object” can reference to implementation-specific parameters
+7. “Object” is the consumer-facing declaration for workloads
+8. Cluster “Default” “Object” is the network the cluster has been created with
+9. Cluster “Default” “Object” cannot be changed or removed
+10. Pods shall reference the “object” when trying to attach to the specific
+networks
+    * Pods has to specify all the “object” that it wishes to attach to,
+      including the Cluster “Default” “Object”
+11. Workloads can access “Object” when “attach” RBAC exists at the attachment time
+    * “attach” RBAC is new verb to allow referencing API objects
+12. The Pod reference to a Networks is optional and when NOT specified, Pod
+connects to “Default” “Object” (network the cluster has been created with)
+13. Pod shall be able to provide additional configuration on how it attaches to
+a network
+    * Identify what “Object” is the Primary network
+    * Optional parameters: MAC address, IP address, speed, MTU, interface name etc.
+14. Every Pod connected to specific network (represented by the “Object”) must
+have connectivity within that network between each other within the Cluster
+    * A Pod connected to specific network (represented by the “Object”) may or
+      may not have cross connectivity between different networks (represented by
+      the “Object”)
+15. Pods attached to a network are connected to each other in a manner defined
+by the “Object” implementation
+16. Basic network Interface information for each attachment will be exposed to
+runtime Pod (via e.g. environment variables, downward API etc.)
+
+#### Phase II (scheduler, kubelet and API probing)
```

Comment:
Would QoS and resource handling fit here? Since this phase is concerned with scheduler, maybe it should cover resource management too - if network would be in the same category as CPU or memory, it would suggest using the same scheduling/pooling mechanism

Reply:
I think QoS and "network-resourcing" as a whole would be a new requirement. I would consider this as an future extension.

Reply:
Shouldn't this be "implementation specific"?  aka something your CNI plugin(s) deal with?